xen/drivers/passthrough/coproc.c

```diff
@@ -0,0 +1,87 @@
+/*
+ * Code to provide SCF ability to manipulate IOMMU
+ *
+ * Oleksandr Tyshchenko <Oleksandr_Tyshchenko@epam.com>
+ * Copyright (C) 2017 EPAM Systems Inc.
+ *
+ * This program is free software; you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation; either version 2 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ */
+
+#include <xen/lib.h>
+#include <xen/sched.h>
+#include <xen/guest_access.h>
+#include <xen/iommu.h>
+#include <xsm/xsm.h>
+
+#include "../../arch/arm/coproc/coproc.h"
+
+int iommu_assign_coproc(struct domain *d, device_t *dev)
+{
+    const struct domain_iommu *hd = dom_iommu(d);
+    int rc;
+
+    if ( !iommu_enabled || !hd->platform_ops ||
```

Comment:
No, it doesn't. Nowhere in existing Xen code I don't see attempts to protect it. It being set at early stage only at remains untouched